apps/web/src/components/providers/EnvironmentProvider.tsx

```diff
@@ -69,15 +69,19 @@ function selectEnvironment(environments: IEnvironment[] | undefined | null, sele
 export function EnvironmentProvider({ children }: { children: React.ReactNode }) {
   const navigate = useNavigate();
   const queryClient = useQueryClient();
-  const { currentOrganization } = useAuth();
+  const { currentOrganization, isLoading: isLoadingAuth } = useAuth();
+  const [internalLoading, setInternalLoading] = useState(true);
   const {
     data: environments,
-    isLoading,
+    isLoading: isLoadingEnvironments,
     refetch: refetchEnvironments,
   } = useQuery<IEnvironment[]>([QueryKeys.myEnvironments, currentOrganization?._id], getEnvironments, {
     enabled: !!currentOrganization,
     retry: false,
     staleTime: Infinity,
+    onSettled: (data, error) => {
```

Comment:
<!-- Notes are always non-blocking and simply highlight something the reader should take note of.  -->
🗒 **note (non-blocking):** ‏ `onSettled` seems [to be deprecated](https://tanstack.com/query/v4/docs/framework/react/reference/useQuery) due to unstable behaviour 

maybe `onSuccess` would be sufficient replacement?

Reply:
but I also want it to trigger on error, so an alternative would be to have both the onSuccess and onFailure handler. But the hooks need to be